xtask/src/main.rs

```diff
@@ -91,6 +94,30 @@ fn execute_task() -> Result<()> {
 
 fn install_tools() -> Result<()> {
     println!("Installing project tools...");
+
+    const PINNED_NIGHTLY: &str = "nightly-2025-09-01";
+
+    let rustup_steps = [
+        // Installe la toolchain sans les composants inutiles
+        ["toolchain", "install", "--profile", "minimal", PINNED_NIGHTLY],
+        // Ajoute rustfmt pour cette toolchain
+        ["component", "add", "rustfmt", "--toolchain", PINNED_NIGHTLY],
+    ];
+
+    for args in rustup_steps {
+        let status = std::process::Command::new("rustup")
+            .args(args)
+            .status()
+            .expect("Failed to execute rustup");
```

Comment:
Please don't use `expect` (or `unwrap`), instead return the `Result` back with `?`. @egithinji, can you review this part? We should (eventually) have some useful patterns in the `xtask` code to easily and quickly execute shell commands.

Reply:
Ah, I see! Let's update the existing code in a different PR to keep this focused on your changes.